src/database/test/DatabaseTests.cpp

```diff
@@ -349,3 +355,430 @@ TEST_CASE("schema test", "[db]")
+        auto const tlOptLiabilities = {
+            make_optional<Liabilities>(Liabilities{0, 6}),
+            nullopt<Liabilities>(),
+            make_optional<Liabilities>(Liabilities{0, 0}),
+            make_optional<Liabilities>(Liabilities{5, 8}),
+            nullopt<Liabilities>()};
+
+        // Pair up each of the optional liabilities in tlOptLiabilities with a
+        // new valid account.
+        AccOptLiabilitiesVec accOptLiabilitiesVec;
+        std::transform(accOptLiabilities.begin(), accOptLiabilities.end(),
+                       std::back_inserter(accOptLiabilitiesVec),
+                       [](OptLiabilities const& aol) {
+                           return AccOptLiabilitiesPair(
+                               LedgerTestUtils::generateValidAccountEntry(),
+                               aol);
+                       });
+
+        // Pair up each of the optional liabilities in tlOptLiabilities with a
+        // new valid trustline.
+        TLOptLiabilitiesVec tlOptLiabilitiesVec;
+        std::transform(tlOptLiabilities.begin(), tlOptLiabilities.end(),
+                       std::back_inserter(tlOptLiabilitiesVec),
+                       [](OptLiabilities const& tlol) {
+                           return TLOptLiabilitiesPair(
+                               LedgerTestUtils::generateValidTrustLineEntry(),
+                               tlol);
+                       });
+
+        // Create a data entry to test that its extension and ledger
+        // entry extension have the default (empty-XDR-union) values.
+        auto de = LedgerTestUtils::generateValidDataEntry();
+
+        // Create an offer entry to test that its extension and ledger
+        // entry extension have the default (empty-XDR-union) values.
+        auto oe = LedgerTestUtils::generateValidOfferEntry();
+
+        // Create the application, with the code above that inserts old-schema
+        // accounts and trustlines into the database injected between database
+        // creation and upgrade.
+        Config const& cfg = getTestConfig(0, dbMode);
+        VirtualClock clock;
+        Application::pointer app =
+            createTestApplication<SchemaUpgradeTestApplication,
+                                  SchemaUpgradeTestApplication::PreUpgradeFunc>(
+                clock, cfg,
+                [prepOldSchemaDB, accOptLiabilitiesVec, tlOptLiabilitiesVec, de,
+                 oe](SchemaUpgradeTestApplication& sapp) {
+                    prepOldSchemaDB(sapp, accOptLiabilitiesVec,
+                                    tlOptLiabilitiesVec, de, oe);
+                });
+        app->start();
+
+        // Validate that the accounts and trustlines have the expected
+        // liabilities and ledger entry extensions now that the database upgrade
+        // has completed.
+
+        LedgerTxn ltx(app->getLedgerTxnRoot());
+
+        for (auto aol : accOptLiabilitiesVec)
+        {
+            LedgerKey key;
```

Comment:
Use the `LedgerEntryKey` function (applies in multiple places).

Reply:
Done.